directories/src/redux/epics/node-epics.js

```diff
@@ -143,9 +147,34 @@ const requestGenesisHashEpic = (action$, store) => {
             brokerNode.completeGenesisHashPoW(txid, trytesArray[0])
           ).map(({ data }) => {
             const { purchase: genesisHash, numberOfChunks } = data;
-            return nodeActions.addGenesisHash({ genesisHash, numberOfChunks });
-          })
+            return { genesisHash, numberOfChunks };
+          }).catch(error => {
+            console.log("GENESIS HASH COMPLETE ERROR", error);
+            return Observable.empty();
+          });
         )
+        .mergeMap(({ genesisHash, numberOfChunks }) => {
+          const datamap = Datamap.generate(genesisHash, numberOfChunks);
+          const addresses = _.values(datamap);
+          const randomNumber = AppUtils.randomNumber(addresses.length);
+          const randomSectorAddresses = addresses.slice(randomNumber);
+          return Observable.fromPromise(
+             iota.findTransactions(randomSectorAddresses)
+          ).map(({ transaction }) => {
+            const { bundles, addresses, tags, approvees } = transaction;
+            return { bundles, addreses, tags, approvees, genesisHash, numberOfChunks };
```

Comment:
nit:  should be "addresses" not "addreses"

Also, the output of findTransactions should just be an array of transaction hashes, but it looks like you're expecting something different?

Reply:
Yeah, thats wrong.. thanks, will repair it.. we will get array of hashes

https://iota.readme.io/reference#findtransactions